lib/array_intersection.rb

```diff
@@ -1,4 +1,22 @@
+# time complexity => O(n + m)
+# Space complexity => O(n)
 def intersection(list1, list2)
-  raise NotImplementedError, "Intersection not implemented"
+  lookup_hash = {}
+  result = []
+  list1.each do |num|
+    if lookup_hash[num]
+      lookup_hash[num] = false
+    else
+      lookup_hash[num] = true
+    end
```

Comment:
This works as the input lists are without duplicates, but you can instead just do this.

```suggestion
    lookup_hash[num] = true
```